packages/ui-tree-browser/src/components/TreeBrowser/TreeButton/index.js

```diff
@@ -114,7 +132,33 @@ export default class TreeButton extends Component {
     }
   }
 
-  render () {
+  renderFulltext() {
+    const { name, descriptor } = this.props
+
+    return (
+      <>
+        <header>{name}</header>
+        {descriptor && <span>{descriptor}</span>}
+      </>
+    )
+  }
+
+  elementIsTruncated(el) {
+    if (!el) {
+      return false
+    }
+
+    return el.offsetWidth < el.scrollWidth
```

Comment:
Using TruncateText to handle truncation in TreeButtons would give you access to an `onUpdate` callback. The first argument of is a boolean that lets you know whether or not the content is currently truncated. This would handle this kind of logic for you.